qiskit/providers/ibmq/runtime/ibm_runtime_service.py

```diff
@@ -540,6 +567,28 @@ def _decode_job(self, raw_data: Dict) -> RuntimeJob:
                           params=decoded,
                           creation_date=raw_data.get('created', None))
 
+    def _encode_program_data(self, data: Union[bytes, str]) -> bytes:
+        """Encodes a runtime program data into Base64. Program data can be
+        either a filepath (str) or raw bytes.
+
+        Args:
+            data: the program data
+
+        Raises:
+            FileNotFoundError: If the data filepath does not exist.
+
+        Returns:
+            the encoded data
+        """
+        # Convert any file data into bytes
+        if isinstance(data, bytes):
+            return data
+        with open(data, 'r') as file:
+            data = file.read().encode()
+        # Convert the bytes data into Base64
+        data = base64.b64encode(data)
```

Comment:
@rathishcholarajan Does this seem like it covers what you are looking for?
```suggestion
        # If the input is bytes, encode it.
        if isinstance(data, bytes):
            return base64.b64encode(data)
        # If the input is a str, read the file it points to and encode it.
        with open(data, 'r') as file:
            return base64.b64encode(file.read().encode())
```

Reply:
Do we need to do `file.read().encode()`? Wouldn't `file.read()` suffice? Also I'd prefer something like this which is a bit more concise. (Don't have to have multiple return statements).

```
if isinstance(data, str):
    with open(data, 'r') as file:
        data = file.read()
return base64.b64encode(data)
```

Reply:
Updated suggestion! Note: `base64.b64encode(..)` only accepts bytes, so a string would have to be encoded.

Reply:
Thanks. I think you have to indent the `with` block inside the `if` block? Also doesn't `file.read()` by itself return bytes?

Reply:
Updated! `read()` returns just a string though, when I run that block without `encode()`, it throws an error when encoding into base64: 
```encoded = binascii.b2a_base64(s, newline=False)
TypeError: a bytes-like object is required, not 'str'

Reply:
Aah ok if used 'rb' then file.read() returns bytes. This is fine. Thanks.